### PR TITLE
✨(dimail) automate allows requests to dimail when creating domain accesses

### PR DESCRIPTION
**Pull request by mjeammet**
## Purpose

Automatically send user creation and allow creation requests to dimail
upon creating domain access.

## Proposal

Description...

- [x] client function to send user creation request
- [x] client fuction to send allow creation request
- [x] test basic case for admin/owner
- [x] test basic case for viewer (no call made)
- [x] test wen user already exists
- [x] make sure it triggers when creating a domain !
- [ ] handle update of existing domain access + test OK when user goes from viewer to admin/owner (in another PR ?)

### CODE REVIEW

**Comment by mjeammet** [comment]
Something it breaking test `test_api_mail_domain__accesses_create_authenticated_administrator` : the `validate` method should raise an exception line 177 and prevent us from entering the `create`. Still, it seems like it doesn't stop us.

Do you have any idea what is happening, @sdemagny ?

**Comment by mjeammet** [comment]
@qbey e2e tests are failing because of a navigation action that doesn't work as expected. Could it be due to the fact that, upon logging on localhost:3000 as people@people, I land on /teams/ and `mail-domains` tab icon has disappeared ? 
I feel like you mentioned something like that the other day.

**Comment by mjeammet** [comment]
Reproduction en local de ce qui casse le test e2e : 
image

File: src/backend/core/api/client/viewsets.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def list(self, request, *args, **kwargs):
             # - are profile contacts for a user
             user__isnull=True,
             # - are overriden base contacts
-            overridden_by__isnull=True,
+            overriding_contacts__isnull=True,
```
Comment (mjeammet) [inline src/backend/core/api/client/viewsets.py:155]: @Morendil c'est quoiiii çaaaaaa ?

**Review by Morendil — APPROVED** [review]

**Pull request merged**